tests/integration/conftest.py

```diff
@@ -0,0 +1,112 @@
+import logging
+import pathlib
+import socket
+import tempfile
+import time
+
+import pytest
+import trio
+
+
+@pytest.fixture
+def dopple_file():
+    return pathlib.Path(__file__).parent.parent.parent / "dopple" / "dopple.py"
+
+
+def generate_docker_cmds(name, cmd_continuation):
+    """
+    Generate ``start`` and ``stop`` commands for any given cmd that is intended to use
+    docker under the current host's user account.
+    """
+    start_cmd = (
```

Comment:
Running the different clients through docker allows us to test against a wide range of clients without going through any installation hassle.